README.md

```diff
@@ -1,46 +1,64 @@
-# Advanced Sample Hardhat Project
-
-This project demonstrates an advanced Hardhat use case, integrating other tools commonly used alongside Hardhat in the ecosystem.
-
-The project comes with a sample contract, a test for that contract, a sample script that deploys that contract, and an example of a task implementation, which simply lists the available accounts. It also comes with a variety of other tools, preconfigured to work with the project code.
-
-Try running some of the following tasks:
-
-```shell
-npx hardhat accounts
-npx hardhat compile
-npx hardhat clean
-npx hardhat test
-npx hardhat node
-npx hardhat help
-REPORT_GAS=true npx hardhat test
-npx hardhat coverage
-npx hardhat run scripts/deploy.ts
-TS_NODE_FILES=true npx ts-node scripts/deploy.ts
-npx eslint '**/*.{js,ts}'
-npx eslint '**/*.{js,ts}' --fix
-npx prettier '**/*.{json,sol,md}' --check
-npx prettier '**/*.{json,sol,md}' --write
-npx solhint 'contracts/**/*.sol'
-npx solhint 'contracts/**/*.sol' --fix
+# About
+The Typescript library for developers to use with Medusa-enabled applications
+
+# Installation
+
+```bash
+npm install --save @medusa-network/medusa-sdk
 ```
 
-# Etherscan verification
+# Usage
 
-To try out Etherscan verification, you first need to deploy a contract to an Ethereum network that's supported by Etherscan, such as Ropsten.
+[See the demo application for more examples of usage](https://github.com/medusa-network/medusa-app)
 
-In this project, copy the .env.example file to a file named .env, and then edit it to fill in the details. Enter your Etherscan API key, your Ropsten node URL (eg from Alchemy), and the private key of the account which will send the deployment transaction. With a valid .env file in place, first deploy your contract:
+```typescript
+import { Medusa, EVMG1Point, SuiteType } from "@medusa-network/medusa-sdk";
 
-```shell
-hardhat run --network ropsten scripts/deploy.ts
-```
+// Initialize medusa with:
+// - The address of the Medusa Encryption Oracle contract
+// - A signer for a user
+const signer = new ethers.Wallet(userPrivateKey).connect(ethers.getDefaultProvider());
+const medusaOracleAddress = "0xabc...123"
+const medusa = await Medusa.init(medusaOracleAddress, signer);
+// Note: Medusa can be initialized one or more times for a given encryption suite
+
+// Get Public Key of Medusa Oracle contract
+const medusaPublicKey = await medusa.fetchPublicKey()
+
+// Prompt a user to sign a message with their wallet and derive their medusa keypair from their (deterministic) signature
+const keypair = await medusa.sign();
```

Comment:
sign is usually meant to return a signature, that's confusing as well here. Maybe `medusa.keypairFromWallet()`? Idk what's the best name, maybe what you had before as well (`deriveKeyPair`IIRC) was not so bad. Naming is hard ><

Reply:
Yeah I wanted to call it `signIn` or `signin` but it looked weird imo. Because the method returns the keypair, but it also saves it as a member variable inside the `medusa` instance (so it doesn't have to be passed to encrypt/decrypt later)